src/test/java/org/jenkinsci/plugins/workflow/cps/DSLTest.java

```diff
@@ -314,6 +314,70 @@ public void namedSoleParamForStep() throws Exception {
         p.setDefinition(new CpsFlowDefinition("try {def c = classLoad(getClass().name); error(/did not expect to be able to load ${c} from ${c.classLoader}/)} catch (ClassNotFoundException x) {echo(/good, got ${x}/)}", false));
         r.assertBuildStatusSuccess(p.scheduleBuild2(0));
     }
+
+    @Test public void userDefinedClosureInvocationExecution() throws Exception {
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        p.setDefinition(new CpsFlowDefinition("binding[\"my_closure\"] = { \n" +
+                                              " sleep 1 \n" + 
+                                              " echo \"my closure!\" \n" + 
+                                              "}\n" + 
+                                              "my_closure() ", false));
+        WorkflowRun b = r.assertBuildStatusSuccess(p.scheduleBuild2(0));
+        r.assertLogContains("my closure!", b); 
+    }
+
+    @Test public void userDefinedClosureShellInvocationExecution() throws Exception {
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        p.setDefinition(new CpsFlowDefinition("my_closure = { \n" +
+                                              " sleep 1 \n" + 
+                                              " node{ \n" + 
+                                              "   sh \"pwd\" \n" + 
```

Comment:
This will fail tests under Windows agents -- please see some of the other examples that use the Hudson utility function to determine if we're on Windows vs. Unix and run the appropriate step (sh or bad).